_topics/en/op_checktemplateverify.md

```diff
@@ -0,0 +1,105 @@
+---
+title: OP_CHECKTEMPLATEVERIFY
+
+## Optional.  An entry will be added to the topics index for each alias
+aliases:
+  - OP_CHECKOUTPUTSHASHVERIFY
+
+## Required.  At least one category to which this topic belongs.  See
+## schema for options
+categories:
+  - Scripts and Addresses
+  - Soft Forks
+  - Fee Management
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **OP_CHECKTEMPLATEVERIFY (CTV)** is a proposed new opcode that takes a
+  commitment hash as a parameter and requires any transaction executing
+  the opcode include a set of outputs that match the commitment.  This
+  makes it possible to create an address that specifies how any funds
+  received to that address may be spent---a design known in Bitcoin as a
+  *covenant*.
+
+## TODO: link to [topic vaults] when available
+## Optional.  Use Markdown formatting.  Multiple paragraphs.  Links allowed.
+extended_summary: |
+  Originally introduced under the name **OP_CHECKOUTPUTSHASHVERIFY**
+  (COSHV), the proposal initially focused on the ability to create
+  *congestion control transactions* where a spender pays a single
+  address using CTV which, when confirmed to a suitable depth, then
+  assures several receivers that they can each be paid.  This two-step
+  process can probably be used anywhere payment batching is an option
+  but it can likely reduce fees even further than payment batching.
+
+  Later versions of the proposal placed greater emphasis on other
+  contracts and [covenants][topic covenants] that could be created using
+  the new opcode, such as the ability to create [channel
+  factories][topic channel factories], vaults, and
+  [coinjoin transactions][topic coinjoin] in new ways that might
+  simplify construction or reduce fees.  Other authors have mentioned
+  that the new opcode could possibly be used to allow users to
+  trustlessly [pool their funds][joinpool] together into a single UTXO
+  in a way that would increase privacy.
+
+  Criticisms of the proposal have focused on it being [too
+  specific][maxwell-attributed specific] to the congestion control
+  use case rather than providing a [flexible][corallo flex] and
+  [generic][oconnor generic] covenant capability.
+
+  [maxwell-attributed specific]: https://bitcointalk.org/index.php?topic=5220520.msg53711340#msg53711340
+  [corallo flex]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-May/016936.html
```

Comment:
I suggest we drop this link. I think the Russell O'Connor critique on flexibility and genericness is much more constructive and informative.